cluster/examples/kubernetes/cassandra/operator.yaml

```diff
@@ -0,0 +1,227 @@
+                  required:
+                    - "name"
+                    - "members"
+                    - "storage"
+                    - "resources"
+              required:
+                - "name"
+          required:
+            - "version"
+            - "datacenter"
+
+---
+
+## Cassandra Member CRD
+#apiVersion: apiextensions.k8s.io/v1beta1
+#kind: CustomResourceDefinition
+#metadata:
+#  name: members.cassandra.rook.io
+#spec:
+#  group: cassandra.rook.io
+#  names:
+#    kind: Member
+#    listKind: MemberList
+#    plural: members
+#    singular: member
+#  scope: Namespaced
+#  version: v1alpha1
+
+---
+
+# ClusterRole for cassandra-operator.
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: rook-cassandra-operator
+rules:
+  - apiGroups:
+    - ""
+    resources:
+      - pods
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - ""
+    resources:
+      - services
+      - serviceaccounts
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+  - apiGroups:
+      - rbac.authorization.k8s.io
+    resources:
+      - roles
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+  - apiGroups:
+      - rbac.authorization.k8s.io
+    resources:
+      - rolebindings
+    verbs:
+      - get
+      - list
+      - watch
+      - create
+  - apiGroups:
+      - apps
+    resources:
+      - statefulsets
+    verbs:
+      - "*"
+  - apiGroups:
+      - policy
+    resources:
+      - poddisruptionbudgets
+    verbs:
+      - create
+  - apiGroups:
+      - cassandra.rook.io
+    resources:
+      - "*"
+    verbs:
+      - "*"
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - create
+      - update
+      - patch
+---
+# ServiceAccount for cassandra-operator. Serves as its authorization identity.
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: rook-cassandra-operator
+  namespace: rook-cassandra-system
+---
+# Bind cassandra-operator ServiceAccount with ClusterRole.
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: rook-cassandra-operator
+  namespace: rook-cassandra-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rook-cassandra-operator
+subjects:
+- kind: ServiceAccount
+  name: rook-cassandra-operator
+  namespace: rook-cassandra-system
+---
+# cassandra-operator StatefulSet.
+ apiVersion: apps/v1
+ kind: StatefulSet
+ metadata:
+   name: rook-cassandra-operator
+   namespace: rook-cassandra-system
+ spec:
+   replicas: 1
+   serviceName: "non-existent-service"
+   selector:
+     matchLabels:
+       app: rook-cassandra-operator
+   template:
+     metadata:
+       labels:
+         app: rook-cassandra-operator
+     spec:
+       serviceAccountName: rook-cassandra-operator
+       containers:
+       - name: rook-cassandra-operator
+         image: "192.168.1.8:5000/rook/cassandra:master"
```

Comment:
this will need to be updated to point at the public docker hub registry before going to master